src/main/java/com/google/javascript/cl2dts/DeclarationGenerator.java

```diff
@@ -513,10 +513,32 @@ public Void caseTemplatizedType(TemplatizedType type) {
             emit("[]");
             return null;
           }
+          Iterator<JSType> it = type.getTemplateTypes().iterator();
+          if ("Object".equals(referencedType.getReferenceName())) {
```

Comment:
indeed, `typeRegistry.getNativeType(JSTypeNative.OBJECT_TYPE)`, fixed here and for Array above